corehq/apps/data_interfaces/interfaces.py

```diff
@@ -78,3 +81,156 @@ def report_context(self):
+
+
+class BulkArchiveFormInterface(SubmitHistoryMixin, DataInterface, ProjectReport):
+    name = ugettext_noop("Manage Forms")
+    slug = "bulk_archive_forms"
+
+    report_template_path = 'data_interfaces/interfaces/archive_forms.html'
+
+    def __init__(self, request, **kwargs):
+        super(BulkArchiveFormInterface, self).__init__(request, **kwargs)
+        self.fields = self.fields + ['corehq.apps.data_interfaces.interfaces.ArchiveOrNormalFormFilter']
+        self.mode = FormManagementMode(request.GET.get('archive_or_restore'))
+
+    @property
+    def template_context(self):
+        context = super(BulkArchiveFormInterface, self).template_context
+        context.update({
+            "form_query_string": self.request.GET.urlencode(),
+            "mode": self.mode,
+            "total_xForms": int(self.es_results['hits']['total']),
+        })
+        return context
+
+    def _es_xform_filter(self):
+        return self.mode.xform_filter
+
+    @property
+    def headers(self):
+        h = [
+            DataTablesColumn(
+                mark_safe(
+                    """
+                    Select  <a class="select-visible btn btn-mini btn-inverse">all</a>
+                    <a class="select-none btn btn-mini btn-warning">none</a>
+                    """
+                ),
+                sortable=False, span=3
+            ),
+            DataTablesColumn(_("View Form"), span=2),
+            DataTablesColumn(_("Username"), prop_name='form.meta.username', span=3),
+            DataTablesColumn(
+                _("Submission Time") if self.by_submission_time
+                else _("Completion Time"),
+                prop_name=self.time_field,
+                span=3,
+            ),
+            DataTablesColumn(_("Form"), prop_name='form.@name'),
+        ]
+        h.extend([DataTablesColumn(field) for field in self.other_fields])
+        return DataTablesHeader(*h)
+
+    @property
+    def rows(self):
+        submissions = [res['_source'] for res in self.es_results.get('hits', {}).get('hits', [])]
+
+        for form in submissions:
+            display = FormDisplay(form, self)
+            checkbox = mark_safe(
+                """<input type="checkbox" class="xform-checkbox"
+                value="{form_id}" name="xform_ids"/>"""
+            )
+            yield [
+                checkbox.format(form_id=form["_id"]),
+                display.form_data_link,
+                display.username,
+                display.submission_or_completion_time,
+                display.readable_form_name,
+            ] + display.other_columns
+
+    @property
+    def form_ids_response(self):
+        from corehq.elastic import es_query
+        from corehq.pillows.mappings.xform_mapping import XFORM_INDEX
+
+        results = es_query(
+            params={'domain.exact': self.domain},
+            q=self.filters_as_es_query(),
+            es_url=XFORM_INDEX + '/xform/_search',
+        )
```

Comment:
Seems like you should pass in `fields=['_id']` to avoid fetching the entire form